src/olympia/addons/tests/test_models.py

```diff
@@ -1102,60 +1102,72 @@ def test_newlines_less_than_tight(self):
+            FIREFOX_EXT_CATS['social-communication']
+        ]
+
+        addon = get_addon()
+        assert set(addon.all_categories) == set(expected_firefox_cats)
+        for cat in addon.all_categories:
             assert cat.application == amo.FIREFOX.id
 
-        cats = [c24, c23, c22]
-        app_cats = [(amo.FIREFOX, cats)]
-        assert addon().app_categories == app_cats
+        assert addon.app_categories == {amo.FIREFOX: expected_firefox_cats}
 
-        c = Category(application=amo.THUNDERBIRD.id, db_name='XXX',
-                     type=addon().type, count=1, weight=1)
-        c.save()
-        AddonCategory.objects.create(addon=addon(), category=c)
-        c24.save()  # Clear the app_categories cache.
-        app_cats += [(amo.THUNDERBIRD, [c])]
-        assert addon().app_categories == app_cats
+        thunderbird_static_cat = (
+            CATEGORIES[amo.THUNDERBIRD.id][amo.ADDON_EXTENSION]['tags'])
+        category = Category.from_static_category(thunderbird_static_cat)
+        category.save()
+        AddonCategory.objects.create(addon=addon, category=category)
+        # Reload the addon to get a fresh, uncached categories list.
+        addon = get_addon()
+        assert addon.app_categories == {
+            amo.FIREFOX: expected_firefox_cats,
+            amo.THUNDERBIRD: [thunderbird_static_cat]
+        }
+        assert set(addon.all_categories) == set(
+            expected_firefox_cats + [thunderbird_static_cat])
 
-    def test_app_categories_sunbird(self):
+    def test_app_categories_ignore_unknown_cats(self):
         def get_addon():
             return Addon.objects.get(pk=3615)
 
         addon = get_addon()
 
-        # This add-on is already associated with three Firefox categories.
-        cats = sorted(addon.categories.all(), key=lambda x: x.name)
-        assert addon.app_categories == [(amo.FIREFOX, cats)]
-
-        # Associate this add-on with a Sunbird category.
-        c2 = Category.objects.create(application=amo.SUNBIRD.id,
+        # This add-on is already associated with three Firefox categories
+        # using fixtures: Bookmarks, Feeds, Social.
+        FIREFOX_EXT_CATS = CATEGORIES[amo.FIREFOX.id][amo.ADDON_EXTENSION]
+        expected_cats = [
+            FIREFOX_EXT_CATS['bookmarks'],
+            FIREFOX_EXT_CATS['feeds-news-blogging'],
+            FIREFOX_EXT_CATS['social-communication']
+        ]
+        assert addon.app_categories == {amo.FIREFOX: expected_cats}
+
+        # Associate this add-on with a couple more categories, including
+        # one that does not exist in the constants.
+        c2 = Category.objects.create(application=amo.SUNBIRD.id, id=123456,
                                      type=amo.ADDON_EXTENSION, name='Sunny D')
         AddonCategory.objects.create(addon=addon, category=c2)
-
-        # Sunbird category should be excluded.
-        assert get_addon().app_categories == [(amo.FIREFOX, cats)]
+        static_thunderbird_category = (
```

Comment:
same as test above
 (if there were more than 2 tests I'd say make it a separate class and move the setup to `setUp` but meh)